src/controllers/confirm.company.controller.ts

```diff
@@ -16,11 +21,29 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = async (req: Request, res: Response, next: NextFunction) => {
+export const post = async (req: Request, res: Response) => {
+  const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+  const session: Session = req.session as Session;
+  const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, company.companyNumber);
+  if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
+    await createNewConfirmationStatement(req);
+    const url = TRADING_STATUS_PATH.replace(":companyNumber", company.companyNumber);
+    return res.redirect(url);
+  } else {
+    return displayEligibilityStopPage(res, eligibilityStatusCode );
+  }
+};
+
+const displayEligibilityStopPage = (res: Response, eligibilityStatusCode: EligibilityStatusCode) => {
+  if (eligibilityStatusCode === EligibilityStatusCode.INVALID_COMPANY_STATUS) {
```

Comment:
line 38 isn't fully covered by the tests. Need to test when the status code is not INVALID_COMPANY_STATUS.
It looks like It will cause the page to hang if that is the case as nothing will get returned. Need to cover off that scenario - perhaps log an error with the unhandled EligibilityStatusCode and throw an error that will be caught in the post try catch which will then show the error page by calling next(e)